src/log/log.c

```diff
@@ -2832,3 +2783,68 @@ __wt_log_flush(WT_SESSION_IMPL *session, uint32_t flags)
         WT_RET(__wt_log_force_sync(session, &lsn));
     return (0);
 }
+
+/*
+ * __log_fsync_dir --
+ *     Perform fsync of log->log_dir_fh. Requires log->log_sync_lock to be held by the caller.
+ */
+int
+__log_fsync_dir(WT_SESSION_IMPL *session, WT_LSN *min_lsn, const char *method)
+{
+    WT_LOG *log;
+    uint64_t fsync_duration_usecs, time_start, time_stop;
+
+    log = S2C(session)->log;
+
+    if (log->sync_dir_lsn.l.file < min_lsn->l.file) {
+        WT_ASSERT(session, log->log_dir_fh != NULL);
+        __wt_verbose(session, WT_VERB_LOG, "%s: sync directory %s to LSN %" PRIu32 "/%" PRIu32,
+          method, log->log_dir_fh->name, min_lsn->l.file, min_lsn->l.offset);
+        time_start = __wt_clock(session);
+        WT_RET(__wt_fsync(session, log->log_dir_fh, true));
+        time_stop = __wt_clock(session);
+        fsync_duration_usecs = WT_CLOCKDIFF_US(time_stop, time_start);
+        WT_ASSIGN_LSN(&log->sync_dir_lsn, min_lsn);
+        WT_STAT_CONN_INCR(session, log_sync_dir);
+        WT_STAT_CONN_INCRV(session, log_sync_dir_duration, fsync_duration_usecs);
+    }
+
+    return (0);
+}
+
+/*
+ * __log_fsync_file --
+ *     use a new file handle to perform the fsync.
+ */
+int
```

Comment:
Same here. Should be static, and move them together.

Also add that this must be called with the sync lock held.

Reply:
Done (looks like I deleted lines before committing :()